### PR TITLE
Add functions to child theme

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This adds some functionality to the child theme's `functions.php`. This breaks down into two basic categories:

1. Functions that are called by page templates (first commit)
1. Functions that get called automatically during the WordPress page lifecycle (second commit)

### Secrets

- [ ] All new secrets have been added to Pantheon tiers
- [ ] Relevant secrets have been updated in Github Actions
- [ ] All new secrets documented in README
- [ ] No new secrets are defined

### Documentation

- [ ] Project documentation has been updated
- [ ] No documentation changes are needed

### Accessibility

- [ ] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)

### Stakeholder approval

- [ ] Stakeholder approval has been confirmed
- [ ] Stakeholder approval is not needed

### Dependencies

NO dependencies are updated

## Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] The changes have been verified
- [ ] The documentation has been updated or is unnecessary
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

**Comment by matt-bernhardt** [comment]
The codeclimate errors are because CodeClimate doesn't completely respect that `phpcs:disable` and `phpcs:enable` directive.

File: web/app/themes/mitlib-child/functions.php [hunk 1]
```diff
@@ -32,3 +32,127 @@ function child_scripts_styles() {
...
+ */
+function register_child_nav() {
+	register_nav_menu( 'child-nav', 'Child Nav' );
+}
+add_action( 'init', 'Mitlib\Child\register_child_nav' );
+
+/**
+ * Remove parent theme page templates.
+ *
+ * The Parent theme includes a number of page templates which are meant for only
```
Comment (JPrevost) [inline web/app/themes/mitlib-child/functions.php:87]: Hmm. I wonder if this is a sign that they shouldn't be in the parent theme. I'm absolutely not suggesting we try to make a change now to our theme hierarchy but this feels like a code smell.

**Review by JPrevost — APPROVED** [review]

File: web/app/themes/mitlib-child/functions.php (hunk 1 above)
Comment (matt-bernhardt) [reply web/app/themes/mitlib-child/functions.php:87]: Hmm... that's an interesting notion. I'd like to discuss this further. I agree that there's an aspect of code smell here, but the alternatives feel icky too.

Comment (matt-bernhardt) [reply web/app/themes/mitlib-child/functions.php:87]: https://mitlibraries.atlassian.net/browse/LM-199 is now in the backlog to have this discussion

**Pull request merged**